test/e2e/framework/prometheus.go

```diff
@@ -168,3 +170,53 @@ func (f *Framework) DeletePrometheusAndWaitUntilGone(name string) error {
 func promImage(version string) string {
 	return fmt.Sprintf("quay.io/prometheus/prometheus:%s", version)
 }
+
+func (f *Framework) WaitForTargets(amount int) error {
+	var targets []*Target
+
+	if err := f.Poll(time.Minute*10, time.Second, func() (bool, error) {
+		var err error
+		targets, err = f.GetActiveTargets()
+		if err != nil {
+			return false, err
+		}
+
+		if len(targets) == amount {
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("Waiting for targets timed out. %v of %v targets found. %v", len(targets), amount, err)
+	}
+
+	return nil
+}
+
+func (f *Framework) GetActiveTargets() ([]*Target, error) {
+	resp, err := http.Get(fmt.Sprintf("http://%s:30900/api/v1/targets", f.ClusterIP))
```

Comment:
this is ok for now but we can keep in the back of our heads that we should rather proxy these requests when possible rather than exposing on node port